Web Inspector: Remove undocked-title-area element since inspector already adjusts based on safeAreaInsets

the-chenergy opened the pull request:
#### afe7a8d8ea3366ef7a6870e27b5329645cfb956a
<pre>
Web Inspector: Remove undocked-title-area element since inspector already adjusts based on safeAreaInsets
<a href="https://rdar.apple.com/149898806">rdar://149898806</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=292205">https://bugs.webkit.org/show_bug.cgi?id=292205</a>

Reviewed by Devin Rousso and BJ Burg.

The patch <a href="https://github.com/WebKit/WebKit/pull/43779">https://github.com/WebKit/WebKit/pull/43779</a> made Web Inspector
automatically adjust its visible area based on the WKInspectorWKWebView&apos;s
safeAreaInsets, which natively computes the margins on all four edges
that are blocked by other view elements. That patch introduced a
regression that the undocked Web Inspector started to have an extra
white space on the top under the title bar, which we then fixed in <a href="https://github.com/WebKit/WebKit/pull/44438.">https://github.com/WebKit/WebKit/pull/44438.</a>
However, it turns out safeAreaInsets already includes the title bar&apos;s
height, so there is actually no need for the undocked-title-area element
anymore as the auto-adjustment already does the job well.

Note that dragging the title bar to move the window is an effect handled
by the window natively. By removing the undocked-title-area and thus a
call to WI.moveUndockedWindowMouseDown, we&apos;re letting the window itself
handle the drag to move instead. This preserves the normal dragging
behavior, and I tested that manually.

* Source/WebInspectorUI/UserInterface/Base/Main.js:
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Views/Main.css:
(#main):
(#undocked-title-area): Deleted.
(body.mac-platform #undocked-title-area): Deleted.
(body:not(.mac-platform) #undocked-title-area): Deleted.
(body.docked #undocked-title-area): Deleted.
(body.window-inactive #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body.mac-platform #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body:not(.mac-platform) #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body.mac-platform.window-inactive #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body:not(.mac-platform).window-inactive #undocked-title-area): Deleted.
* Source/WebInspectorUI/UserInterface/Views/Popover.js:
* Source/WebInspectorUI/UserInterface/Views/TabBar.css:
(.tab-bar):
* Source/WebInspectorUI/UserInterface/Views/Variables.css:
(:root):
(&amp;.mac-platform:not(.docked)): Deleted.
   - Deprecate and remove the undocked-title-area element.

* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.h:
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webView]):
(-[WKInspectorViewController observeValueForKeyPath:ofObject:change:context:]):
(-[WKInspectorViewController webViewWebContentProcessDidTerminate:]):
(-[WKInspectorViewController setIsAttached:]): Deleted.
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
(WebKit::WebInspectorUIProxy::platformAttach):
(WebKit::WebInspectorUIProxy::platformDetach):
   - Restore the auto-adjusting behavior to also still take effect while
     inspector is undocked, to let that handle the padding required for
     the title bar&apos;s height.

Canonical link: <a href="https://commits.webkit.org/294303@main">https://commits.webkit.org/294303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5df790cc32e04568b20c04c8bbc3f2ca33b67c68

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21030 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11335 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106519 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51995 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21338 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29524 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77206 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34244 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104373 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16464 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91562 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57547 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16284 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9573 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51343 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86165 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9648 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108871 "Hash 5df790cc for PR 44623 does not build (failure)") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28495 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20968 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/108871 "Hash 5df790cc for PR 44623 does not build (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28857 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87764 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/108871 "Hash 5df790cc for PR 44623 does not build (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30466 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8181 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22612 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16505 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28426 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33705 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28237 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31557 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29796 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg commented on Source/WebInspectorUI/UserInterface/Base/Main.js:
```diff
@@ -2068,18 +2063,12 @@ WI._handleDockedResizerMouseDown = function(event)
 
-WI._handleUndockedTitleAreaMouseDown = function(event)
-{
-    WI.moveUndockedWindowMouseDown(event);
```
It's still possible to drag the undocked inspector by its titlebar, right? I can't tell just by looking at the code, but it does seem `WI.TabBar` does also call `WI.moveUndockedWindowMouseDown` but I'm not sure whether the drag event would be handled by the window properly.

When I tried deleting the undocked title area myself, it seems like the \<body\> element takes some of the hit target away from the TabBar.

Whichever way it is, please explain in the changelog how to test this in both docked and undocked scenarios. Please include title bar dragging as well as adjustments to dock/undocking.

the-chenergy replied:
Good concern. Dragging the undocked inspector is unaffected by this change at all: where from the top we start rendering Web Inspector's contents does not prevent the title bar itself from natively handling dragging, even if there are contents rendered underneath on the Z axis. We're not relying on `WI.moveUndockedWindowMouseDown` to be called in order to move the window.

The changes in the backend ensures we don't render stuff underneath the title bar (on the Z axis) to leave that area blank. That also has no effect on the native dragging behavior.

I'll clarify in the change log that native dragging is preserved like normal.

dcrousso replied:
along those lines, if it's actually not possible to drag the window from the `WI.TabBar` anymore then id say we get rid of `WI.moveUndockedWindowMouseDown` (as well as `InspectorFrontendHost.startWindowDrag`/`InspectorFrontendHost.moveWindowBy`/etc.)

dcrousso replied:
oh also can you upload before and after screenshots in light mode and dark mode when undocked so that we can ensure the UI looks the same?

dcrousso commented on Source/WebInspectorUI/UserInterface/Base/Main.js:
```diff
@@ -476,11 +476,6 @@ WI.contentLoaded = function()
     }
 
-    if (supportsUndocked) {
-        let undockedTitleAreaElement = document.getElementById("undocked-title-area");
```
might be worth checking with other platforms to make sure this wont cause issues

im pretty sure it's fine since for everything other than macOS `--undocked-title-area-height: 0px;` but couldn't hurt to make sure

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/TabBar.css:
```diff
@@ -26,7 +26,6 @@
 .tab-bar {
     display: flex;
     position: absolute;
-    top: var(--undocked-title-area-height);
```
just to be safe this should probably still have `top: 0`

the-chenergy committed:
Web Inspector: Remove undocked-title-area element since inspector already adjusts based on safeAreaInsets
rdar://149898806
https://bugs.webkit.org/show_bug.cgi?id=292205

Reviewed by Devin Rousso and BJ Burg.

The patch https://github.com/WebKit/WebKit/pull/43779 made Web Inspector
automatically adjust its visible area based on the WKInspectorWKWebView's
safeAreaInsets, which natively computes the margins on all four edges
that are blocked by other view elements. That patch introduced a
regression that the undocked Web Inspector started to have an extra
white space on the top under the title bar, which we then fixed in https://github.com/WebKit/WebKit/pull/44438.
However, it turns out safeAreaInsets already includes the title bar's
height, so there is actually no need for the undocked-title-area element
anymore as the auto-adjustment already does the job well.

Note that dragging the title bar to move the window is an effect handled
by the window natively. By removing the undocked-title-area and thus a
call to WI.moveUndockedWindowMouseDown, we're letting the window itself
handle the drag to move instead. This preserves the normal dragging
behavior, and I tested that manually.

* Source/WebInspectorUI/UserInterface/Base/Main.js:
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Views/Main.css:
(#main):
(#undocked-title-area): Deleted.
(body.mac-platform #undocked-title-area): Deleted.
(body:not(.mac-platform) #undocked-title-area): Deleted.
(body.docked #undocked-title-area): Deleted.
(body.window-inactive #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body.mac-platform #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body:not(.mac-platform) #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body.mac-platform.window-inactive #undocked-title-area): Deleted.
(@media (prefers-color-scheme: dark) body:not(.mac-platform).window-inactive #undocked-title-area): Deleted.
* Source/WebInspectorUI/UserInterface/Views/Popover.js:
* Source/WebInspectorUI/UserInterface/Views/TabBar.css:
(.tab-bar):
* Source/WebInspectorUI/UserInterface/Views/Variables.css:
(:root):
(&.mac-platform:not(.docked)): Deleted.
   - Deprecate and remove the undocked-title-area element.

* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.h:
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webView]):
(-[WKInspectorViewController observeValueForKeyPath:ofObject:change:context:]):
(-[WKInspectorViewController webViewWebContentProcessDidTerminate:]):
(-[WKInspectorViewController setIsAttached:]): Deleted.
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
(WebKit::WebInspectorUIProxy::platformAttach):
(WebKit::WebInspectorUIProxy::platformDetach):
   - Restore the auto-adjusting behavior to also still take effect while
     inspector is undocked, to let that handle the padding required for
     the title bar's height.

Canonical link: https://commits.webkit.org/294303@main
